pkg/controller/drain.go

```diff
@@ -179,7 +185,23 @@ func (o *DrainOptions) RunDrain() error {
 		return err
 	}
 
-	err := o.deleteOrEvictPodsSimple()
+	node, err := o.client.CoreV1().Nodes().Get(o.nodeName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		klog.V(4).Infof("Node %q not found, skipping drain", o.nodeName)
+		return nil
+	} else if err != nil {
+		klog.Errorf("Error getting details for node: %q. Err: %v", o.nodeName, err)
+		return err
+	}
+
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status != corev1.ConditionTrue && (time.Since(condition.LastTransitionTime.Time) > o.nodeNotReadyDuration) {
+			klog.Warningf("Skipping drain for NotReady Node %q", o.nodeName)
+			return nil
+		}
```

Comment:
Can we return `err` to get it reconciled again if the machine has been `NotReady` for less than 5 mins?  
- Basically, if node is `NotReady` then check 
     - if more than 5 mins: skip drain
     - if less than 5 mins: return err and let it be reconciled again. If it heals itself then we are lucky else after 5 mins worth of reconciles, the machine drain will be skipped.

WDYT ?

Thanks to @ggaurav10 for the suggestion.

Reply:
Hm, I'll take this comment back, there seems to be another undesirable side-effect of returning `err` here, especially when the node is flipping between ready/not-ready. 

Pointed out by @ggaurav10